tools/cli/internal/versioning/version.go

```diff
@@ -0,0 +1,94 @@
+// Copyright 2024 MongoDB Inc
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//	http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+package versioning
+
+import (
+	"fmt"
+	"regexp"
+	"time"
+)
+
+type APIVersion struct {
+	version     string
+	versionDate time.Time
+}
+
+const (
+	dateFormat = "2006-01-02"
+)
+
+// NewAPIVersion creates a new API version.
+func NewAPIVersionFromDateString(version string) (*APIVersion, error) {
+	versionDate, err := NewVersionDate(version)
+	if err != nil {
+		return nil, err
+	}
+
+	return &APIVersion{
+		version:     version,
+		versionDate: versionDate,
+	}, nil
+}
+
+// NewAPIVersionFromContentType creates a new API version from a content type of the expected format.
+func NewAPIVersionFromContentType(contentType string) (*APIVersion, error) {
```

Comment:
I see a lot of these `NewFrom` methods consider a functional constructor
`New(withDate())`

We do something like this in the store of the CLI 
https://github.com/mongodb/mongodb-atlas-cli/blob/bf07d65101b3707ee25ba0f04d1ae375d5a2d465/internal/store/store.go#L298-L310

Check also how we group multiple options for "easy" defaults

Reply:
oooo i like this, thank you

Reply:
done, this is much better, thanks gus!